http/client.go

```diff
@@ -531,6 +525,42 @@ func (c *InternalClient) marshalImportValuePayload(index, field string, shard ui
 	return buf, nil
 }
 
+// ImportRoaring fast import of raw bits in roaring standard format
+func (c *InternalClient) ImportRoaring(ctx context.Context, node *pilosa.Node, index, field string, shard uint64, data []byte) error {
+	if index == "" {
+		return pilosa.ErrIndexRequired
+	} else if field == "" {
+		return pilosa.ErrFieldRequired
+	}
+
+	endpoint := fmt.Sprintf("/index/%s/field/%s/import-roaring/%d", index, field, shard)
+
+	// Create URL.
+	u := nodePathToURL(node, endpoint)
+	// Generate HTTP request.
+	req, err := http.NewRequest("POST", u.String(), bytes.NewBuffer(data))
+	if err != nil {
+		return errors.Wrap(err, "creating request")
+	}
+	req.Header.Set("Accept", "application/x-binary")
+	req.Header.Set("User-Agent", "pilosa/"+pilosa.Version)
+
+	// Execute request against the host.
+	resp, err := c.httpClient.Do(req.WithContext(ctx))
+	if err != nil {
+		return errors.Wrap(err, "executing request")
+	}
+	defer resp.Body.Close()
+
+	// Validate status code.
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("invalid status: %d", resp.StatusCode)
+	}
+	io.Copy(ioutil.Discard, resp.Body)
```

Comment:
err... this should probably reading the body and returning an error `if resp.Err != ""`, right?

Reply:
yep, my first guess was just only http errors where sent back, but i was wrong

Reply:
yes, good catch